src/lib/mina.ts

```diff
@@ -773,3 +805,143 @@ function defaultNetworkState(): NetworkValue {
     nextEpochData: cloneCircuitValue(epochData),
   };
 }
+
+async function verifyAccountUpdate(
+  account: LedgerAccount,
+  accountUpdate: AccountUpdate,
+  transactionCommitments: {
+    commitment: Field;
+    fullCommitment: Field;
+  },
+  proofsEnabled: boolean
+): Promise<void> {
+  let perm = account.permissions;
+
+  let { commitment, fullCommitment } = transactionCommitments;
+
+  // we are essentially only checking if the update is empty or an actual update
+  function includesChange(val: any): boolean {
+    if (val?.constructor === Array) {
+      return !(val as Array<any>).every((v) => v == null);
+    } else {
+      return val != null;
```

Comment:
it's a tiny nit, but can we in general be explicit about the distinction between `null` and `undefined`? The JSON types all have `null` for indicating a missing field:
https://github.com/o1-labs/snarkyjs/blob/f1a6b131d6bcc66003fa640e10e9e9ddfa3145dd/src/snarky/gen/transaction-json.ts#L32
(maybe more general nit: can we use `===` instead of `==` always if we don't _specifically_ want to handle the cases where `==` does casting?)